sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2909,11 +2909,11 @@ object GpuOverrides {
       "Collect a list of non-unique elements, not supported in reduction",
       // GpuCollectList is not yet supported in Reduction context.
       ExprChecks.aggNotReduction(
-        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 +
+        TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 + TypeSig.NULL +
             TypeSig.STRUCT + TypeSig.ARRAY + TypeSig.MAP),
         TypeSig.ARRAY.nested(TypeSig.all),
         Seq(ParamCheck("input",
-          (TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 +
+          (TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 + TypeSig.NULL +
```

Comment:
What is `TypeSig.NULL`? Does that mean a nullable column?

Reply:
It is Spark's `NullType` which is the datatype representing NULL values.  This is a separate concept from a nullable type.  For example, you might be parsing CSV data with an inferred schema, and Spark could infer the schema is `NullType` because it only contains null values and thus it couldn't deduce a specific type beyond that.

Reply:
NullType really only shows up as a corner case of SQL, but we do some abusive things in our tests. If I run `select null as a_col from table` what is the type of `a_col`? For Spark it is a NullType because it could not infer any other type from it.  If I did `select CAST(null as int) as a_col from table` now spark knows it is an int.